kernel/aster-nix/src/time/timer.rs

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::{
+    boxed::Box,
+    collections::BinaryHeap,
+    sync::{Arc, Weak},
+    vec::Vec,
+};
+use core::time::Duration;
+
+use aster_frame::{arch::register_interrupt_callback, sync::SpinLock};
+
+use super::{clock::Clock, timer_callback::TimerCallback};
+
+/// A timer with periodic functionality.
+///
+/// Setting the timer will trigger a callback function upon expiration of
+/// the set time. If the timer has a interval time set, it will use
+/// the interval time to configure a new timing after expiration.
+pub struct IntervalTimer {
+    interval: SpinLock<Duration>,
+    timer_manager: TimerManager,
+    callback_function: Box<dyn Fn() + Send + Sync>,
+    timer_callback: SpinLock<Weak<TimerCallback>>,
+}
```

Comment:
Well, something confuses me is that `TimerManager` creates `Timer`, while `IntervalTimer` is built upon `TimerManager`. There are two timers under different levels, `IntervalTimer` and `Timer`. Then, I can't understand the timer_callback is for which timer. Could you please make them look different in names?

Reply:
There is only a kind of Timer, which is `IntervalTimer`. `TimerManager` is used to create the corresponding `IntervalTimer`.